### DISCUSSION TITLE
How to share the configuration globally ?

### DISCUSSION THREAD

**Discussion by TheUncharted** [facebookresearch/hydra#2357] [Q&A]
Hi

What is the best practice / simple way for sharing configuration through all the application with Hydra ?
For example I have a project and multiple files that may use a variables from the configuration

Do you use the decorator to get a config object that you inject everywhere ?

**Comment by Jasha10** [comment]
> Do you use the decorator to get a config object that you inject everywhere ?

Yes, this is what I do.

Once you use the decorator (or the compose API) to get a config object, use normal Python idioms to share it between components of your application. Some common patterns are:
- inject / pass around parts of the config object as needed
- use a global variable or singleton pattern to store the config object

Generally programs with global state are harder to reason about and maintain, so for most cases I recommend the first option above.
